utils/keys_test.go

```diff
@@ -298,20 +401,153 @@ func TestImportKeys2InOneFile(t *testing.T) {
 
 	in := bytes.NewBuffer(byt)
 
-	err := ImportKeys(in, []Importer{s})
+	err := ImportKeys(in, []Importer{s}, "", "", passphraseRetriever)
 	require.NoError(t, err)
 
 	bFinal, bRest := pem.Decode(s.data["ankh"])
 	require.Equal(t, b.Bytes, bFinal.Bytes)
-	require.Len(t, bFinal.Headers, 0) // path header is stripped during import
+	_, ok := bFinal.Headers["path"]
+	require.False(t, ok, "expected no path header, should have been removed at import")
+	role, _ := bFinal.Headers["role"]
+	require.Equal(t, notary.DefaultImportRole, role)
 
 	b2Final, b2Rest := pem.Decode(bRest)
 	require.Equal(t, b2.Bytes, b2Final.Bytes)
-	require.Len(t, b2Final.Headers, 0) // path header is stripped during import
+	_, ok = b2Final.Headers["path"]
+	require.False(t, ok, "expected no path header, should have been removed at import")
 	require.Len(t, b2Rest, 0)
 
 	cFinal, cRest := pem.Decode(s.data["morpork"])
 	require.Equal(t, c.Bytes, cFinal.Bytes)
-	require.Len(t, cFinal.Headers, 0)
+	_, ok = cFinal.Headers["path"]
+	require.False(t, ok, "expected no path header, should have been removed at import")
 	require.Len(t, cRest, 0)
 }
+
+func TestImportKeys2InOneFileNoPath(t *testing.T) {
+	s := NewTestImportStore()
+
+	from, _ := os.OpenFile("../fixtures/secure.example.com.key", os.O_RDONLY, notary.PrivKeyPerms)
+	defer from.Close()
+	fromBytes, _ := ioutil.ReadAll(from)
+	b, _ := pem.Decode(fromBytes)
+	b.Headers["role"] = data.CanonicalSnapshotRole
+	bBytes := pem.EncodeToMemory(b)
+
+	b2, _ := pem.Decode(fromBytes)
+	b2.Headers["role"] = data.CanonicalSnapshotRole
+	b2Bytes := pem.EncodeToMemory(b2)
+
+	c := &pem.Block{
+		Headers: make(map[string]string),
+	}
+	c.Bytes = make([]byte, 1000)
+	rand.Read(c.Bytes)
+	c.Headers["path"] = "morpork"
+
+	bBytes = append(bBytes, b2Bytes...)
+	cBytes := pem.EncodeToMemory(c)
+
+	byt := append(bBytes, cBytes...)
+
+	in := bytes.NewBuffer(byt)
+
+	err := ImportKeys(in, []Importer{s}, "", "", passphraseRetriever)
+	require.NoError(t, err)
+
+	bFinal, bRest := pem.Decode(s.data[filepath.Join(notary.NonRootKeysSubdir, "12ba0e0a8e05e177bc2c3489bdb6d28836879469f078e68a4812fc8a2d521497")])
```

Comment:
Non-blocking question, just because I don't know what we decided for this:  it seems like we are ok with one of the base non-root roles not having a gun also (we don't delete it, but we don't error if we can't infer one) - we infer a path being just the non-root subdirectory + the key ID, as you've included in your test.

I'm ok with that, since the user messed this up, but this would mean that earlier versions of docker probably wouldn't be able to use this key?  cc @riyazdf @endophage